core/src/test/resources/org/apache/calcite/test/RelOptRulesTest.xml

```diff
@@ -17792,6 +17792,31 @@ LogicalProject(EMPNO=[$0], ENAME=[$1], JOB=[$2], MGR=[$3], HIREDATE=[$4], SAL=[$
   LogicalSort(sort0=[$7], sort1=[$0], dir0=[ASC], dir1=[ASC])
     LogicalProject(EMPNO=[$0], ENAME=[$1], JOB=[$2], MGR=[$3], HIREDATE=[$4], SAL=[$5], COMM=[$6], DEPTNO=[$7], SLACKER=[$8], EXPR$9=[null:NULL])
       LogicalTableScan(table=[[CATALOG, SALES, EMP]])
+]]>
+    </Resource>
+  </TestCase>
+  <TestCase name="testSortRemoveDuplicateKeys">
+    <Resource name="sql">
+      <![CDATA[select d1
+ from (select deptno as d1, deptno as d2 from dept) as tmp
```

Comment:
do you want to add a test with NULLS FIRST or something similar?

Reply:
This issue hasn't been fully understood yet. Duplicate keys should originate from the same expression, and their NULLS FIRST property should be identical. This rule only considers the removal of duplicate keys.

Reply:
well, if you sort a second time by the same key, even with a different direction, it should not change the result

Reply:
Do you mean that I extend the existing test to `order by d1, d2, d3 desc`, and then ensure that it is `order by deptno` in the end?

Reply:
I think `ORDER BY x ASC, x DESC` is the same as `ORDER BY x ASC`

Reply:
Yes, do I need to extend the test case?

Reply:
Got it. I will extend the test later.